src/libraries/Native/Unix/System.Native/pal_process.c

```diff
@@ -203,6 +202,39 @@ static int SetGroups(uint32_t* userGroups, int32_t userGroupsLength, uint32_t* p
     return rv;
 }
 
+static int32_t SocketPair(int32_t sv[2])
+{
+    int32_t result;
+
+    int type = SOCK_STREAM;
+#ifdef SOCK_CLOEXEC
+    type |= SOCK_CLOEXEC;
+#endif
+
+    while ((result = socketpair(AF_UNIX, type, 0, sv)) < 0 && errno == EINTR);
+
+#ifndef SOCK_CLOEXEC
+    if (result == 0)
+    {
+        while ((result = fcntl(sv[0], F_SETFD, FD_CLOEXEC)) < 0 && errno == EINTR);
+        if (result == 0)
+        {
+            while ((result = fcntl(sv[1], F_SETFD, FD_CLOEXEC)) < 0 && errno == EINTR);
+        }
+
+        if (result != 0)
+        {
+            int tmpErrno = errno;
+            close(sv[0]);
+            close(sv[1]);
```

Comment:
Instead of closing `sv[0]` and `sv[1]`, set them to `-1` so that `CloseIfOpen()` can actually close them.  Otherwise, this is a race condition and unrelated file descriptors can be closed.

Reply:
> set them to -1

Why would we want to set them to -1?  That would overwrite the fds, and CloseIfOpen would have nothing to close.  Don't you mean just removing this if block entirely?

Reply:
Oh, yes, you're right. Yes, removing this `if` here would fix the race condition. (Maybe add a comment in its place?)